### PR TITLE
Authorisation

### PR DESCRIPTION
**Pull request by harture**
Implementation of authorization enforcement

### CODE REVIEW

File: api/management/api.go [hunk 1]
```diff
@@ -12,6 +12,7 @@ type UserRepresentation struct {
 	Label         *string `json:"label,omitempty"`
 	Gender        *string `json:"gender,omitempty"`
 	BirthDate     *string `json:"birthDate,omitempty"`
+	Groups         *[]string `json:"group,omitempty"`
```
Comment (fperot74) [inline api/management/api.go]: This line seems not to be aligned like others... Formatting issue ?

File: cmd/keycloakb/keycloak_bridge.go [hunk 2]
```diff
@@ -151,6 +152,24 @@ func main() {
 		errc <- fmt.Errorf("%s", <-c)
 	}()
 
+	// Authorizations
+	var authorizations management.Authorizations
+	{
+		json, err := ioutil.ReadFile(authorizationConfigFile)
+
+		if err != nil {
+			logger.Log("msg", "could not read YAML authorisation file", "error", err)
```
Comment (fperot74) [inline cmd/keycloakb/keycloak_bridge.go]: Typo: authorization ?

File: cmd/keycloakb/keycloak_bridge.go [hunk 3]
```diff
@@ -151,6 +152,24 @@ func main() {
...
+			return
+		}
+
+		authorizations, err = management.LoadAuthorizations(string(json))
+
+		if err != nil {
+			logger.Log("msg", "could not load authorisations", "error", err)
```
Comment (fperot74) [inline cmd/keycloakb/keycloak_bridge.go]: Typo: authorization ?

File: pkg/management/authorisation.go [hunk 4]
```diff
@@ -0,0 +1,342 @@
...
+
+		if wildcard || realmAllowed || (targetRealm != "master" && nonMasterRealmAllowed) {
+			return nil
+		}
+	}
+
+	return ForbiddenError{}
+}
+
+// ForbiddenError when an operation is not permitted.
```
Comment (fperot74) [inline pkg/management/authorisation.go]: If we use something like this below sample code, we can avoid adding a case in managementErrorHandler() for each new error and avoid unit tests.
```
var ForbiddenError = HTTPError{
	Status:  http.StatusForbidden,
	Message: "ForbiddenError: Operation not permitted.",
}
```
or less 'pretty' but more generic, something like the CreateForbiddenError(...) like the existing CreateMissingParameterError(...)

File: api/management/api.go (hunk 1 above)
Comment (harture) [reply api/management/api.go]: Done

File: cmd/keycloakb/keycloak_bridge.go (hunk 2 above)
Comment (harture) [reply cmd/keycloakb/keycloak_bridge.go]: Done

File: cmd/keycloakb/keycloak_bridge.go (hunk 3 above)
Comment (harture) [reply cmd/keycloakb/keycloak_bridge.go]: Done

**Pull request merged**